tasks/request_status_for_granule/request_status_for_granule.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, Any, List
+
+import database
+from cumulus_logger import CumulusLogger
+from requests_db import get_dbconnect_info, DatabaseError
+
+GRANULE_ID_KEY = 'granule_id'
+ASYNC_OPERATION_ID_KEY = 'asyncOperationId'
+
+LOGGER = CumulusLogger()
+
+
+def task(granule_id: str, async_operation_id: str = None) -> Dict[str, Any]:
+    """
+
+    Args:
+        granule_id: The unique ID of the granule to retrieve status for.
+        async_operation_id: An optional additional filter to get a specific job's entry.
+    Returns:
+        todo
+    """
+    if granule_id is None:
+        raise ValueError(f"{GRANULE_ID_KEY} cannot be None.")
+    pass
+
+
+def get_status_entries_for_granule(granule_id: str, async_operation_id: str = None) -> List[Dict[str, Any]]:
+    """
+    Gets the orca_recoverfile status entry for the associated granule_id.
+    If async_operation_id is non-None, then it will be used to filter results.
+    Otherwise, only the item with the most recent request_time will be returned.
+
+    Args:
+        granule_id: The unique ID of the granule to retrieve status for.
+        async_operation_id: An optional additional filter to get a specific job's entry.
+    Returns: todo
+    """
+    raise NotImplementedError
+    # todo: Move this code into a new requests_db.py when able.
+    # todo: Do we join in restore_destination/request_time/completion_time, or just get the whole orca_recoveryjob?
+    # todo: The above may affect what we get from orca_recoverfile, as some properties may be redundant.
+    sql = """
+            SELECT
```

Comment:
You may want to use grouping and json aggregates so the output is in a ready made format so you don't have to parse it. Check out:
- https://www.postgresql.org/docs/9.6/functions-json.html
- https://hashrocket.com/blog/posts/faster-json-generation-with-postgresql

Using common table expressions may make it easier https://www.postgresql.org/docs/9.6/queries-with.html